test/Conversion/tritongpu_to_llvm.mlir

```diff
@@ -606,36 +624,29 @@ module attributes {"triton_gpu.num-warps" = 4 : i32} {
     %tensor = triton_gpu.alloc_tensor : tensor<2x32x32xf32, #A>
     %index = arith.constant 1 : i32
 
+    // TODO: split CHECK tags below in PTX and GCN specific tags in case they fail for AMD GPU.
```

Comment:
Note, this test was failing for both PTX and AMD targets.

Checks in this test are updated in upstream: https://github.com/openai/triton/blob/main/test/Conversion/tritongpu_to_llvm.mlir#L589-L610

This PR transfers changes from upstream to our fork.